crates/pera-core/src/signature_mpc/mpc_manager.rs

```diff
@@ -33,18 +49,88 @@ impl MPCInstance {
     /// Spawns an asynchronous task to handle incoming messages.
     /// The [`MPCService`] will forward any message related to that instance to this channel.
     fn spawn_mpc_messages_handler(&self, mut receiver: mpsc::Receiver<MPCInput>) {
+        let public_parameters = self.language_public_parameters.clone();
+        let consensus_adapter = Arc::clone(&self.consensus_adapter);
+        let epoch_store = self.epoch_store.clone();
+        let threshold = self.threshold.clone();
         tokio::spawn(async move {
+            let mut party: ProofParty;
             while let Some(message) = receiver.recv().await {
+                match message {
+                    MPCInput::InitEvent(_) => {
+                        party = match Self::handle_mpc_proof_init_event(
+                            public_parameters.clone(),
+                            consensus_adapter.clone(),
+                            epoch_store.clone(),
+                            threshold,
+                        )
+                        .await
+                        {
+                            Ok(party) => party,
+                            Err(err) => {
+                                // This should never happen, as there should be on-chain verification on the init transaction
+                                return;
+                            }
+                        };
+                    }
+                    MPCInput::Message => {
+                        // TODO (#235): Implement MPC messages handling
+                    }
+                }
+
                 // TODO (#235): Implement MPC messages handling
             }
         });
     }
 
-    fn handle_message(&mut self, message: MPCInput) {
+    async fn handle_mpc_proof_init_event(
+        public_parameters: maurer::language::PublicParameters<
+            { maurer::SOUND_PROOFS_REPETITIONS },
+            Lang,
+        >,
+        consensus_adapter: Arc<dyn SubmitToConsensus>,
+        epoch_store: Weak<AuthorityPerEpochStore>,
+        threshold: usize,
+    ) -> anyhow::Result<ProofParty> {
+        let batch_size = 1;
+        let party: ProofParty = proof::aggregation::asynchronous::Party::new_proof_round_party(
+            public_parameters,
+            PhantomData,
+            threshold as group::PartyID,
+            batch_size,
+            &mut OsRng,
+        )?;
+
+        let party: ProofParty = match party.advance(HashMap::new(), &(), &mut OsRng) {
+            Ok(advance_result) => match advance_result {
+                AdvanceResult::Advance((message, new_party)) => {
+                    // It is safe to unwrap here, as the epoch store is already created
+                    let epoch_store = epoch_store.upgrade().unwrap();
```

Comment:
Uhm, where does that happen? :sweat_smile:

Reply:
Your'e right; Added a todo